index.js

```diff
@@ -847,7 +931,7 @@ Lexer.prototype = {
       }
 
       for (var i = 0; i <= str.length; i++) {
-        if (isEndOfAttribute(i)) {
+        if (isEndOfAttribute(i, this)) {
```

Comment:
I think either using a `self` variable or `isEndOfAttribute.call(this, i)` would look cleaner.